cleanup(inventory): Drop support for multiple connections

0xFelix opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:

The support for connections to multiple clusters in the inventory
plugin is dropped to better align with user expectations and how other
inventories work. If inventories of multiple clusters are needed the
inventory can be run multiple times with different configurations.
This also helps to clean up the code and make it simpler.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Based on #114, rebase and merge after it is merged.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
inventory: [Breaking change] Support for multiple connections was dropped
```

jcanocan commented:
Great improvement! Reviewed the PR and I have nothing to comment. Great job!
/lgtm

lyarwood commented:
/approve
/hold

As discussed lets reach out to folks both up and downstream before proceeding with this.

> inventory: Support for multiple connections was dropped

nit - Should we somehow highlight this more as a breaking change?

0xFelix commented:
> nit - Should we somehow highlight this more as a breaking change?

I added the `breaking_changes` label. The release automation will highlight it properly. Also updated the PR release note.

jcanocan commented:
/lgtm

0xFelix commented:
/hold cancel

jcanocan commented:
/lgtm
